src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx

```diff
@@ -1,70 +1,127 @@
+          params: {
+            respondentId,
+            fromDatetime: getDateISO(startDate, startTime),
+            toDatetime: getDateISO(endDate || addDays(startDate, 1), endTime),
+            identifiers: getIdentifiers(filterByIdentifier, identifier, identifiers),
+            versions: versions.map(({ id }) => id),
+          },
+        });
+
+        const decryptedAnswers = result.data.result.map((encryptedAnswer) => {
+          const { userPublicKey, answer, items, itemIds, ...rest } = encryptedAnswer;
+          const decryptedAnswer = getDecryptedActivityData({
+            userPublicKey,
+            answer,
+            items,
+            itemIds,
+          });
+
+          return {
+            answer: decryptedAnswer,
+            items,
+            ...rest,
+          };
+        });
+        setAnswers(decryptedAnswers as unknown as DatavizAnswer[]);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchAnswers();
+  }, [watchFilters]);
 
   return (
-    <StyledReport ref={containerRef}>
-      <StyledHeader isSticky={isHeaderSticky}>
-        <StyledHeadlineLarge color={variables.palette.on_surface}>
-          {activity.name}
-        </StyledHeadlineLarge>
-        <Tooltip tooltipTitle={t('configureServer')}>
-          <span>
-            <StyledTextBtn variant="text" startIcon={<Svg id="export" width="18" height="18" />}>
-              {t('downloadLatestReport')}
-            </StyledTextBtn>
-          </span>
-        </Tooltip>
-      </StyledHeader>
-      <Box sx={{ margin: theme.spacing(4.8, 6.4, 4.8) }}>
-        <FormProvider {...methods}>
-          <ReportFilters minDate={minDate} />
-          <ActivityCompleted
-            responses={activityReport.responses}
-            versions={activityReport.versions}
-          />
-          <Subscales />
-          {activityReport.responseOptions && (
-            <ResponseOptions
-              responseOptions={activityReport.responseOptions}
-              versions={activityReport.versions}
-            />
-          )}
-        </FormProvider>
-      </Box>
-    </StyledReport>
+    <>
+      {isLoading && <Spinner />}
+      <StyledReport ref={containerRef}>
+        <StyledHeader isSticky={isHeaderSticky}>
+          <StyledHeadlineLarge color={variables.palette.on_surface}>
+            {activity.name}
+          </StyledHeadlineLarge>
+          <Tooltip tooltipTitle={t('configureServer')}>
+            <span>
+              <StyledTextBtn variant="text" startIcon={<Svg id="export" width="18" height="18" />}>
+                {t('downloadLatestReport')}
+              </StyledTextBtn>
+            </span>
+          </Tooltip>
+        </StyledHeader>
+        <Box sx={{ margin: theme.spacing(4.8, 6.4) }}>
+          <FormProvider {...methods}>
+            <ReportFilters identifiers={identifiers} versions={versions} />
+            <ActivityCompleted answers={answers} versions={versions} />
+            {/* <Subscales /> */}
```

Comment:
maybe add a TODO for uncomment Subscales after someting (tasks on back-end will be done or other)?